original/zip_image_search.py

```diff
@@ -0,0 +1,178 @@
+    return list_thmbnl_images
+
+
+#this function calls the create_list_unzipped_files function and returns the list of files, in this case, images, from which 
+#we iterate through and detect the faces as a tuple (x1, y1, w, h).
+#It accepts the following parameters: zip file name as file_name_str, and the haar cascade as face_cascade_str
+# It stores the tuples in a key dictionary, the key = img_name, and the value = list of tuples
+#This function takes a LOOOONG time to run :(
+def create_dictionary_of_thmbnls_text (file_name_str, face_cascade_str):
+    
+    list_unzipped_files = create_list_unzipped_files (file_name_str)
+    
+    print ("Starting to create_dictionary_of_thmbnls_text")
+    
+    # loading the face detection classifier
+    face_cascade = cv.CascadeClassifier(face_cascade_str)
+    
+    #iterate through image files and create arrays
+    dictionary = { }
+    
+    for img in list_unzipped_files:
+        img_name = img
+        img = cv.imread(img)
+        gray = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+        array_faces = face_cascade.detectMultiScale(gray, 1.15, 5)
+        array_faces_list = array_faces.tolist()
+        
+        text = pytesseract.image_to_string(img)
+        
+        dictionary [img_name] = [create_list_face_thmbnl_imgs(gray, array_faces_list)] + [text] 
+        
+    # check to see that the dictionary was created correctly
+#     display (dictionary)
+    
+    print ('Done with create_dictionary_of_thmbnls_text!')   
+    
+    return (dictionary)
+
+
+def search_name_return_faces (name_str, file_name_str, face_cascade_str):
+    
+    name_dictionary = create_dictionary_of_thmbnls_text (file_name_str, face_cascade_str)
+    
+    print ("Starting to search_name_return_faces")
+    
+    for key in name_dictionary:
+        text = name_dictionary[key][-1]
+        if (text.find(name_str)) != -1:
+            print ("{} in {} = true".format(name_str, key))
+             #  #create contact sheet from list of images (not perfect, needs work, and the height and width is hard-coded with 5 and 4.
+             # #will need to determine the number of files in a list and probably update the sizes of the thmbnls)
+            count = len(name_dictionary[key][0])
+            
+            ##create a contact sheet from images##
+            display(type(name_dictionary[key][0][0]))
+            first_image=name_dictionary[key][0][0]
+            first_image_copy = first_image.copy()
+            first_image_copy = first_image_copy.thumbnail ([100,100])
+#             first_image.open().show()
+            columns = int(count / 2)
+            rows = int (count / columns)
+            contact_sheet=Image.new(first_image.mode, (first_image.width*columns ,first_image.height*rows))
+            x=0
+            y=0
+
+            # read image(s) and convert to RGB
+            for img in name_dictionary[key][0]:
+                img_copy = img.copy()
+                img_copy = img.thumbnail([100,100])
```

Comment:
I think the copys and thumbnail calls are unneeded. Below we use `contact_sheet.paste(img, (x,y))` so I don't see `img_copy` being used in the program.